server/sonar-server/src/main/java/org/sonar/server/rule/RegisterRules.java

```diff
@@ -202,7 +203,7 @@ private boolean mergeRule(RulesDefinition.Rule def, RuleDto dto) {
     if (mergeDescription(def, dto)) {
       changed = true;
     }
-    if (!dto.getSystemTags().containsAll(def.tags())) {
+    if (!difference(dto.getSystemTags(), def.tags()).isEmpty()) {
```

Comment:
if my understanding is correct, is `def.tags()` has tags not in db, it won't update the db! The `difference(Set,Set)` method works only in one way, not the other